src/test/java/eu/bbmri/eric/csit/service/negotiator/unit/service/PostServiceTest.java

```diff
@@ -29,9 +33,38 @@ public class PostServiceTest {
 
   private AutoCloseable closeable;
 
+  private Post publicPost1;
+
+  private Post publicPost2;
+
+  private Post privatePost1;
+
+  private Post privatePost2;
+
   @BeforeEach
   void before() {
     closeable = MockitoAnnotations.openMocks(this);
+    Negotiation n = new Negotiation();
+    Resource r1 = new Resource();
+    r1.setSourceId("r1");
+    Resource r2 = new Resource();
+    r2.setSourceId("r1");
+    publicPost1 = new Post();
+    publicPost1.setType(PostType.PUBLIC);
+    publicPost1.setNegotiation(n);
+    publicPost1.setText("public post 1");
+    publicPost2 = new Post();
+    publicPost2.setType(PostType.PUBLIC);
+    publicPost2.setNegotiation(n);
+    publicPost2.setText("public post 2");
+    privatePost1 = new Post();
+    privatePost1.setType(PostType.PRIVATE);
+    privatePost1.setResource(r1);
+    privatePost1.setText("private post 1");
+    privatePost2 = new Post();
+    privatePost2.setType(PostType.PRIVATE);
+    privatePost2.setResource(r2);
+    privatePost2.setText("private post 2");
```

Comment:
Please refactor this using a builder and a for loop.

Reply:
Fixed in 5cb22c0